src/main/java/org/jenkinsci/plugins/workflow/steps/ToolStep.java

```diff
@@ -119,7 +126,7 @@ public ListBoxModel doFillNameItems(@QueryParameter String type) {
             String name = step.getName();
             String type = step.getType();
             for (ToolDescriptor<?> desc : ToolInstallation.all()) {
-                if (type != null && !desc.getId().equals(type)) {
+                if (type != null && !desc.getId().equals(type) && !type.equals(SymbolLookup.getSymbolValue(desc))) {
```

Comment:
Very hard to glance at this code and see that it is correct. Make sure you have test coverage for all applicable cases. Probably best to define some dummy `ToolInstallation` types with and without symbols and use `@TestExtension`.